plugins/module_utils/network/sonic/config/snmp/snmp.py

```diff
@@ -0,0 +1,1356 @@
+        """ The command generator when state is merged
+        :rtype: A list
+        :returns: the commands necessary to merge the provided into
+                  the current configuration
+        """
+        test_keys = self.new_test_keys(want)
+        commands = get_diff(want, have, test_keys)
+        request_commands = commands
+        commands_updated = []
+        del_commands = {}
+        delete_user_requests = []
+        requests = []
+        if 'user' in want:
+            delete_user_requests = self.delete_snmp_user_request(commands, have)
+            requests = delete_user_requests
+            delete_commands = get_diff(have, want, test_keys)
+            del_commands = self.get_existing_user(commands, have)
+            user_commands = get_diff(want, delete_commands, test_keys)
+
+            if user_commands:
+                request_commands = user_commands
+
+        requests.extend(self.get_create_snmp_request(request_commands, have, False))
+
+        if commands and len(requests) > 0:
+            if delete_user_requests:
+                commands_updated.extend(update_states(del_commands, "deleted"))
+                commands_updated.extend(update_states(user_commands, "merged"))
+            else:
+                commands_updated.extend(update_states(request_commands, "merged"))
+        else:
+            commands_updated = []
+
+        return commands_updated, requests
+
+    def _state_deleted(self, want, have):
+        """ The command generator when state is deleted
+        :rtype: A list
+        :returns: the commands necessary to remove the current configuration
+                  of the provided objects
+        """
+        requests = []
+        commands = []
+
+        if not have:
+            return commands, requests
+
+        delete_all = False
+
+        if have is None:
+            return commands, requests
+        if not want or want is None:
+            commands = have
+            delete_all = True
+        else:
+            want = self.get_configured_option(want, have)
+            reverse_diff = get_diff(have, want, TEST_KEYS)
+            commands = get_diff(want, reverse_diff, TEST_KEYS)
+
+        requests = self.get_delete_snmp_request(commands, have, delete_all)
+
+        if commands and len(requests) > 0:
+            commands = update_states(commands, "deleted")
+        else:
+            commands = []
+
+        return commands, requests
+
+    def get_configured_option(self, want, config):
+        """ Gets the already configured option when want contains an empty list
+
+        :rtype: A dict
+        :returns: the already configured option
+        """
+        new_config = {}
+        for key, value in want.items():
+            if value == [] and key in config:
+                new_config[key] = config[key]
+            else:
+                new_config[key] = value
+        return new_config
+
+    def delete_snmp_user_request(self, commands, have=None):
```

Comment:
This function seems unnecessary. Please incorporate the "existence" checking done here for "have.get('user') into the function currently called "delete_user' and call that directly from the _state_merged function. Please change the name of the current "delete_user" function to clarify what it does (e.g. delete_snmp_matched_user_config).